crates/transpiler/src/passes/litinski_transformation.rs

```diff
@@ -0,0 +1,218 @@
+            "rz" => {
+                rotations.push(clifford.get_inverse_z(qbits[0]));
+            }
+            _ => panic!("Unsupported gate {}", gate_name),
+        }
+    }
+
+    rotations
+}
+
+#[pyfunction]
+#[pyo3(name = "run", signature = (dag, fix_clifford=true))]
+pub fn run_litinski_transformation(
+    py: Python,
+    dag: &mut DAGCircuit,
+    fix_clifford: bool,
+) -> PyResult<Option<DAGCircuit>> {
+    let op_counts = dag.get_op_counts();
+
+    // Skip the pass if there are no rotation gates.
+    if op_counts
+        .keys()
+        .all(|k| !ROTATION_GATE_NAMES.contains(&k.as_str()))
+    {
+        return Ok(None);
+    }
+
+    // Skip the pass if there are unsupported gates.
+    if !op_counts
+        .keys()
+        .all(|k| SUPPORTED_GATE_NAMES.contains(&k.as_str()))
+    {
+        return Err(TranspilerError::new_err(
+            "Unable to run Litinski tranformation as the circuit contains gates not supported by the pass".to_string()
+        ));
+    }
+
+    let num_qubits = dag.num_qubits();
+
+    // Turn the Qiskit circuit into Rustiq's format: this is a vector of (Pauli string, indices).
+    // Additionally, keep track of the rotation angles, an update to the global phase (produced when
+    // converting T/Tdg gates to RZ-rotations), and Clifford gates in the circuit.
+    let mut rustiq_circuit: Vec<(String, Vec<usize>)> = Vec::new();
+    let mut angles: Vec<Param> = Vec::new();
+    let mut global_phase_update = 0.;
+    let mut clifford_ops: Vec<PackedInstruction> = Vec::new();
+    for node_index in dag.topological_op_nodes()? {
+        if let NodeType::Operation(inst) = &dag[node_index] {
+            let (name, angle, phase_update) = match inst.op.view() {
+                OperationRef::StandardGate(StandardGate::T) => {
+                    ("rz", Some(Param::Float(PI / 8.)), PI / 8.)
+                }
+                OperationRef::StandardGate(StandardGate::Tdg) => {
+                    ("rz", Some(Param::Float(-PI / 8.0)), -PI / 8.)
+                }
+                OperationRef::StandardGate(StandardGate::RZ) => {
+                    let param = &inst.params_view()[0];
+                    ("rz", Some(multiply_param(param, 0.5)), 0.)
+                }
+                _ => (inst.op.name(), None, 0.),
+            };
+
+            global_phase_update += phase_update;
+
+            let qubits: Vec<usize> = dag
+                .get_qargs(inst.qubits)
+                .iter()
+                .map(|q| q.index())
+                .collect();
+
+            rustiq_circuit.push((name.to_string(), qubits));
+
+            if let Some(angle) = angle {
+                // This is a rotation, save the angle.
+                angles.push(angle);
+            } else {
+                // This is a Clifford operation, save it.
+                clifford_ops.push(inst.clone());
+            }
+        } else {
+            unreachable!();
```

Comment:
As a person that has seen program exit with comment-less "unreachable" statements, having a comment makes debugging a whole lot easier 🙂 
```suggestion
            unreachable!("Gate instructions are T/Tdg/RZ at this point.");
```

Reply:
done